expression/expression.go

```diff
@@ -112,25 +112,30 @@ func (e CNFExprs) Clone() CNFExprs {
 }
 
 // EvalBool evaluates expression list to a boolean value.
-func EvalBool(ctx sessionctx.Context, exprList CNFExprs, row chunk.Row) (bool, error) {
+func EvalBool(ctx sessionctx.Context, exprList CNFExprs, row chunk.Row) (bool, bool, error) {
+	isnull := false
 	for _, expr := range exprList {
 		data, err := expr.Eval(row)
 		if err != nil {
-			return false, err
+			return false, false, err
 		}
 		if data.IsNull() {
-			return false, nil
+			isnull = true
```

Comment:
No we cannot, because if the first expr is `null`, while the second expr is `false`, we should return `selected = false, isNull = false, error = nil`, because `null and false` is `false`, not `null`.